src/main.rs

```diff
@@ -475,3 +475,38 @@ fn all_versions(crd: &CustomResourceDefinition) -> String {
     vers.sort_by_cached_key(|v| std::cmp::Reverse(Version::parse(v).priority()));
     vers.join(", ")
 }
+
+fn format_docstr(indent: &str, input: &str) -> String {
+    let rustdoc_code = "```\n";
+    let rustdoc_starts: Vec<usize> = input.match_indices(rustdoc_code).step_by(2).map(|(index, _)| index).collect();
+    let mut cleaned_input = String::with_capacity(input.len());
+    let mut start = 0;
+    for doc in rustdoc_starts {
+        cleaned_input.push_str(&input[start..doc]);
+        cleaned_input.push_str("```text\n");
+        start = doc + rustdoc_code.len();
+    }
+    cleaned_input.push_str(&input[start..]);
+
+    // TODO: maybe logic to split doc strings by sentence / length here
+
+    format!(
+        "{}/// {}",
+        indent,
+        cleaned_input.replace('\n', &format!("\n{}/// ", indent))
+    )
+}
+
+#[cfg(test)]
+mod test {
+    use crate::format_docstr;
+
+    #[test]
+    fn escapes_codes_from_descriptions() {
+        assert_eq!("/// ```text\n/// foobar\n/// ```\n/// ", format_docstr("", "```\nfoobar\n```\n"));
+        assert_eq!("/// Some docs\n/// ```text\n/// foobar\n/// ```\n/// ", format_docstr("", "Some docs\n```\nfoobar\n```\n"));
+        assert_eq!("/// Some docs\n/// ```text\n/// foobar\n/// ```", format_docstr("", "Some docs\n```\nfoobar\n```"));
+        assert_eq!("/// ```text\n/// foobar\n/// ```", format_docstr("", "```\nfoobar\n```"));
+        assert_eq!("/// Some docs\n/// with no code blocks!", format_docstr("", "Some docs\nwith no code blocks!"));
```

Comment:
Nice tests :+1: 

Do they actually run in main? We only run `cargo test --lib` on ci so normally put unit tests outside `main`. Maybe they can be lifted to `output`?

Reply:
I'll check, yeah.

Reply:
Moving "only" the tests? Or should I move the function as well... and then re-export it as it is currently done for other things in `mod output`:

```rust
mod output;
pub use output::{Container, MapType, Member, Output};
```

Maybe making that function `pub(crate)`, tho I see no reason to "hide" it

Reply:
i'm thinking move all of it yeah. can easily export it from `lib.rs` in the same way.